crates/starknet-handler/src/l1_headers_store/l1_headers_store.rs

```diff
@@ -1 +1,108 @@
+use starknet::{
+    accounts::{
+        single_owner::SignError, Account, AccountError, Call, ExecutionEncoding, SingleOwnerAccount,
+    },
+    core::{
+        types::{BlockId, BlockTag, FieldElement, FunctionCall, InvokeTransactionResult},
+        utils::get_selector_from_name,
+    },
+    providers::{jsonrpc::HttpTransport, JsonRpcClient, Provider, ProviderError, Url},
+    signers::LocalWallet,
+};
 
+use crate::util::get_high_and_low;
+
+pub struct L1HeadersStore {
+    provider: JsonRpcClient<HttpTransport>,
+    signer: LocalWallet,
+    l1_headers_store: FieldElement,
+    owner_account: FieldElement,
+}
+
+impl L1HeadersStore {
+    pub fn new(
+        rpc: &str,
+        l1_headers_store: FieldElement,
+        signer: LocalWallet,
+        owner_account: FieldElement,
+    ) -> Self {
+        let url = Url::parse(rpc).unwrap();
+        let provider = JsonRpcClient::new(HttpTransport::new(url));
+
+        Self {
+            provider,
+            signer,
+            l1_headers_store,
+            owner_account,
+        }
+    }
+
+    pub async fn store_state_root(&self, block_number: u64, state_root: String) {
+        let (state_root_high, state_root_low) = get_high_and_low(state_root);
+
+        let entry_point_selector = get_selector_from_name("store_state_root").unwrap();
+        let calldata = vec![
+            FieldElement::from_dec_str(block_number.to_string().as_str()).unwrap(),
+            FieldElement::from_byte_slice_be(&state_root_low.to_be_bytes()).unwrap(),
+            FieldElement::from_byte_slice_be(&state_root_high.to_be_bytes()).unwrap(),
+        ];
+
+        match self.invoke(entry_point_selector, calldata).await {
+            Ok(_) => (),
+            Err(e) => tracing::error!("{:?}", e),
```

Comment:
this function handles potential errors in invoke by logging them, but it can return the errors to the caller instead